### PR TITLE
Add visualization of vectors.

### PR DESCRIPTION
**Pull request by S-Dafarra**
Patched few memory problems related to visualization.

### CODE REVIEW

File: src/visualization/src/IrrlichtUtils.h [hunk 1]
```diff
@@ -160,6 +162,8 @@ inline irr::scene::ISceneNode * addGeometryToSceneManager(const iDynTree::SolidS
 
         geomNode = smgr->addMeshSceneNode(boxMesh,linkNode);
 
+        boxMesh->drop();
```
Comment (traversaro) [inline src/visualization/src/IrrlichtUtils.h]: How did you realized that this was necessary?

File: src/visualization/src/VectorsVisualization.cpp [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    assert(smgr);
+    m_smgr = smgr;
+}
+
+void VectorsVisualization::close()
+{
+    for (auto& vector: m_vectors) {
+        if (vector.visualizationNode) {
+            vector.visualizationNode->removeAll();
+            //vector.visualizationNode->drop();
```
Comment (traversaro) [inline src/visualization/src/VectorsVisualization.cpp]: This can be removed?

File: src/visualization/src/IrrlichtUtils.h (hunk 1 above)
Comment (S-Dafarra) [reply src/visualization/src/IrrlichtUtils.h]: Using valgrind. To be honest the way irrlicht deals with this sort of smart pointers is quite blurry to me. Looking at some examples online I noticed they where dropping the pointer to the mesh after adding it to the scene. As you noticed, I did a lot of experiments before avoiding any problem with valgrind 😁

File: src/visualization/src/VectorsVisualization.cpp (hunk 2 above)
Comment (S-Dafarra) [reply src/visualization/src/VectorsVisualization.cpp]: Yes, it is a leftover of some trial for avoiding memory leaks.

**Comment by S-Dafarra** [comment]
> Cool. Does this means that the visualizer now works in Windows?

Yes! I had to set manually some CMake variables and add some path and it worked. I may add some info in the README maybe.

**Pull request merged**